content/post/2017-09-14-using-ecto-and-sqlite3-with-nerves.md

```diff
@@ -0,0 +1,202 @@
+---
+title: Using Ecto and Sqlite3 with Nerves
+date: 2017-09-14
+author: Connor Rigby
+draft: false
+tags: ["nerves"]
+---
+
+One of the most common questions we answer in the Nerves help channels is how to
+store persistant data accross reboots.
+Since the file system read only, the normal avenues usually will not work with Nerves.
+
+There are several solutions that have yielded varrying levels of success accross
+projects. Before we dive too deep into Sqlite, lets take a look at the other options:
+
+*  Key Value Storage such as the ever popular (Persistant Storage)[https://github.com/cellulose/persistent_storage]
+  * PROS: Super easy setup. Simple api.
+  * CONS: May be too simple. No migrations, File size may be an  issue.
+
+* ETS and DETS
+  * PROS: Built into Erlang. Easy setup. Distributed.
+  * CONS: No migrations system. Can be difficult to maintain.
+
+* Full Databases such as postgres, mongo
+  * PROS: Ecto adapters make these relatively easy, multi user, etc.
+  * CONS: Require a large setup on Nerves - Custom system, configs, setup etc.
+
+And depending on your use case, one or more of those might be more useful to you.
+But I've found in many cases a simple,
+local, non clustered database is the perfect data storage mechanism for an
+embedded system like Nerves.
+
+## Application setup
+Lets walk through a quick example app to get us up and running with Nerves and Ecto+Sqlite3.
+
+```elixir
+mix nerves.new hello_db
+```
+
+First off crack open the `mix.exs` file and add two dependencies to your list.
+```elixir
+# ommited for clarity
```

Comment:
It might be more confusing that this comment is here. Maybe just remove it or change it to:

```elixir
# ...
```